compiler+runtime/src/cpp/jank/runtime/obj/persistent_hash_map.cpp

```diff
@@ -39,6 +39,12 @@ namespace jank::runtime::obj
     this->meta = meta;
   }
 
+  persistent_hash_map::persistent_hash_map(meta_option meta, value_type &&d)
+    : data{ std::move(d) }
+  {
+    this->meta = meta;
```

Comment:
Use this:

```cpp
: data{ std::move(d) }
, meta{ meta }
```

Also please make `meta` const.

Reply:
I get this error which I don't understand:
```
error: member initializer 'meta' does not name a non-static data member or base class
```

Reply:
Oh, `meta` comes in from the base. You'll need something like:

```
: obj::detail::base_persistent_map{ meta }
, data{ std::move(d) }
```

It's possible you need the [full type](https://github.com/jank-lang/jank/pull/184/files/b0b846961ce26929508221e0da368f4dd5f221df#diff-f13107755d1250573233217600a41a5c9f5ae7cb1f20f1dae0cbdd390371c45aR16) there. If so, I'd alias it under `obj_type` called `parent_type`.

We'll need a ctor in `base_persistent_map` which initilializes `meta`.

Reply:
I think https://github.com/jank-lang/jank/pull/184/commits/b892c52db3155523dc412a5664315a2544e56059 is implementing what you described, but it doesn't appear to work:
```clojure
% ./build/jank repl            
Bottom of clojure.core
clojure.core=> (def x ^:meta {:x :y})
#'clojure.core/x
clojure.core=> (meta x)
{:meta true}
clojure.core=> (meta (assoc x :x 1))
nil
clojure.core=> (meta (dissoc x :x))
nil
```

Reply:
Your changes look correct, in that commit. I would either add some logs or use a debugger to see what's going on with the meta here. Using `this->meta = meta;` may work, but that's not the way we do it in C++; we use member initializer lists. Why it's not working for you... I'm not yet sure. Maybe try a `./bin/clean` to be safe.

Reply:
So after getting a debugger setup, I noticed the code paths weren't getting hit and it turns out implementing metadata preservation for `"persistent_hash_map"` isn't useful for `"persistent_array_map"` 😂 After pushing up https://github.com/jank-lang/jank/pull/184/commits/055edf8d5b6d2fb70413936df48f627d09789fd2, things are working. I think my earlier testing was done explicitly with `(hash-map)` and I forgot 🤦

Reply:
Good catch! I didn't spot that either. :grin: